### ISSUE TITLE
Test cases for subscription with real network communication involved

### ISSUE THREAD

**Issue by LingForCC** [ringcentral/ringcentral-csharp#27]
Nowadays we only have test cases which use mock objects. So the subscription related code is not tested formally. I think we should find a way to test it against real network communication. Let's trigger some discussions in this issue to figure out a solution.

**Comment by tylerlong** [comment]
We already have tests cases against real API server: https://github.com/ringcentral/ringcentral-csharp/tree/master/RingCentral.Test.Real

But we don't have any testing code for subscription yet. I am planning to write some code inspired by: http://stackoverflow.com/questions/16449197/unit-testing-a-class-with-event-and-delegate

**Comment by tylerlong** [comment]
I added subscription unit test in this commit: https://github.com/ringcentral/ringcentral-csharp/commit/6b3ed91e01dcc3f7acd258d1969b35643e40bb60


```csharp
var sub = new Subscription.SubscriptionServiceImplementation() { _platform = sdk.Platform };
sub.AddEvent("/restapi/v1.0/account/~/extension/~/message-store");
var called = false;
sub.Subscribe((message) => {
    called = true;
}, null, null);

var dict = new Dictionary<string, dynamic> {
    { "text", "hello world" },
    { "from", new Dictionary<string, string> { { "phoneNumber", Config.Instance.UserName} } },
    { "to", new Dictionary<string, string>[] { new Dictionary<string, string> { { "phoneNumber", Config.Instance.Receiver } } } },
};
var request = new Http.Request("/restapi/v1.0/account/~/extension/~/sms", JsonConvert.SerializeObject(dict));
sdk.Platform.Post(request);

Thread.Sleep(30000);
Assert.AreEqual(true, called);
sub.Remove();
```

I will close this issue as soon as the commit is merged into master branch.

**Comment by grokify** [comment]
Thanks @tylerlong. Closing this since it has been merged into master branch for release `1.0.0-alpha`.
